strings/naive_string_search.py

```diff
@@ -9,24 +11,34 @@
 """
 
 
-def naivePatternSearch(mainString, pattern):
-    patLen = len(pattern)
-    strLen = len(mainString)
-    position = []
-    for i in range(strLen - patLen + 1):
-        match_found = True
-        for j in range(patLen):
-            if mainString[i + j] != pattern[j]:
-                match_found = False
+def naive_pattern_search(main_string: str, pattern: str) -> list:
+    """
+    >>> naive_pattern_search("ABAAABCDBBABCDDEBCABC", "ABC")
+    [4, 10, 18]
+
+    >>> naive_pattern_search("", "ABC")
+    []
+
+    >>> naive_pattern_search("TEST", "TEST")
+    [0]
+
+    >>> naive_pattern_search("ABCDEGFTEST", "TEST")
+    [7]
+    """
+    pat_len, str_len = len(pattern), len(main_string)
+    positions = []
+    for i in range(str_len - pat_len + 1):
+        found = True
+        for j in range(pat_len):
+            if main_string[i + j] != pattern[j]:
+                found = False
                 break
-        if match_found:
-            position.append(i)
-    return position
+
+        if found:
+            positions.append(i)
+
+    return positions
 
 
-mainString = "ABAAABCDBBABCDDEBCABC"
-pattern = "ABC"
-position = naivePatternSearch(mainString, pattern)
-print("Pattern found in position ")
-for x in position:
-    print(x)
+if __name__ == "__main__":
+    assert [3] == naive_pattern_search("ABCDEFG", "DE")
```

Comment:
```suggestion
    assert naive_pattern_search("ABCDEFG", "DE") == [3]
```
Assertions are easier to read if they are in `actual == expected` order.